src/python/pants/backend/python/subsystems/pytest.py

```diff
@@ -54,14 +54,10 @@ class PyTest(PythonToolBase):
     name = "Pytest"
     help_short = "The pytest Python test framework (https://docs.pytest.org/)."
 
-    # Pytest 7.1.0 introduced a significant bug that is apparently not fixed as of 7.1.1 (the most
-    # recent release at the time of writing). see https://github.com/pantsbuild/pants/issues/14990.
-    # TODO: Once this issue is fixed, loosen this to allow the version to float above the bad ones.
-    #  E.g., as default_version = "pytest>=7,<8,!=7.1.0,!=7.1.1"
     default_requirements = [
-        "pytest==7.0.1",
-        "pytest-cov>=2.12,!=2.12.1,<3.1",
-        "pytest-xdist>=2.5,<3",
+        "pytest>=7,<8.4,!=7.1.0,!=7.1.1",
+        "pytest-cov==5.0.0",
+        "pytest-xdist==3.6.1",
```

Comment:
```suggestion
        "pytest>=7,<9,!=7.1.0,!=7.1.1",
        "pytest-cov>=5,<7",
        "pytest-xdist>=3.6.1,<4",
```

And maybe add a comment like this before `default_requirements`:
```python
    # NOTE: We only support using the pinned version in the lockfile.
    # Using version ranges allows the resolver to pick the version to pin, which simplifies manual
    # maintenance (lockfile regen, bump IC, etc). The range should exclude known bad versions, and
    # it should end with the next major unreleased version, or at a major version that is known to
    # introduce changes that are incompatible with how pants uses pytest.
    # latest versions on 2025-08-20: pytest-8.4.1, pytest-cov-6.2.0, pytest-xdist-3.8.1
    default_requirements = [
```

Reply:
I'll think about it, but for now I suspect that https://github.com/pantsbuild/pants/pull/22578 punts the issue.

Generally we should be a lot stricter about requiring the user to specify one interpreter version, and then using it for everything. But that would be some work.